types/underscore/index.d.ts

```diff
@@ -4853,117 +4847,124 @@ declare module _ {
         propertyOf(): (key: string) => any;
 
         /**
-        * Wrapped type `object`.
-        * @see _.isEqual
-        **/
+         * Performs an optimized deep comparison between the wrapped object
+         * and the provided object to determine if they should be considered equal.
+         * @param other Compare to the wrapped object.
+         * @returns True if the wrapped object is equal to `other`.
+         **/
         isEqual(other: any): boolean;
 
         /**
-        * Wrapped type `object`.
-        * @see _.isEmpty
-        **/
+         * Returns true if an enumerable object contains no values (no enumerable own-properties).
+         * For strings and array-like objects checks if the length property is 0.
+         * @returns True if the wrapped object is empty.
+         **/
         isEmpty(): boolean;
 
         /**
-        * Wrapped type `object`.
-        * @see _.isMatch
-        **/
-        isMatch(): boolean;
+         * Tells you if the keys and values in properties are contained in the wrapped object.
+         * @param properties The properties to check for in the wrapped object.
+         * @returns True if all keys and values in `properties` are also in the wrapped object.
+         **/
+        isMatch(properties: any): boolean;
 
         /**
-        * Wrapped type `object`.
-        * @see _.isElement
-        **/
+         * Returns true if the wrapped object is a DOM element.
+         * @returns True if the wrapped object is a DOM element, otherwise false.
+         **/
```

Comment:
As a point of interest, just to see if it would work I tried changing the return type below to `this is `Underscore<never, Element>` and using it as a type guard on a variable containing a wrapped `any`, and while the compiler didn't complain about that and it did have an effect that effect ended up changing the variable into an `Underscore<unknown, any> & Underscore<never, Element>`, which isn't quite right.

Reply:
Why not `this is Underscore<Element, Element>`?

Reply:
That would be appropriate if `Element` was iterable and iterating it yielded `Element` objects, but `_.TypeOfCollection<Element>` results in `never` so that doesn't seem to be the case.

I tried `this is` as an experiment but it doesn't seem to modify the type like I would have expected it to since it does an intersection rather than overriding the type completely. Trying out checking if a `string` is a `number`, though, I found that the resulting type is `never`, so maybe this isn't an unexpected result. Regardless, I feel like the intersection result prevents this from being a useful thing to do since I think to make the result useful one would unfortunately still have to assert the chain's type.

If you think it's useful to do that, though, I'd be happy to update these to modify the type of `this` since it seems to be possible to do so.

Reply:
Honestly, I have no idea. I trust you on this one.

Reply:
I don't think this ends up yielding a terribly useful result unfortunately, and even if it did these functions are probably very rarely used outside of `UnderscoreStatic`, so I think I'll leave this as-is.